src/EnergyPlus/Plant/LoopSide.cc

```diff
@@ -492,30 +492,42 @@ namespace DataPlant {
                 auto const SELECT_CASE_var(ComponentTypeOfNum);
 
                 // possibly air-connected components
-                if ((SELECT_CASE_var == TypeOf_CoilWaterCooling) || (SELECT_CASE_var == TypeOf_CoilWaterDetailedFlatCooling) ||
-                    (SELECT_CASE_var == TypeOf_CoilWaterSimpleHeating) || (SELECT_CASE_var == TypeOf_CoilSteamAirHeating) ||
-                    (SELECT_CASE_var == TypeOf_CoilWAHPHeatingEquationFit) || (SELECT_CASE_var == TypeOf_CoilWAHPCoolingEquationFit) ||
-                    (SELECT_CASE_var == TypeOf_CoilWAHPHeatingParamEst) || (SELECT_CASE_var == TypeOf_CoilWAHPCoolingParamEst) ||
-                    (SELECT_CASE_var == TypeOf_CoilUserDefined) || (SELECT_CASE_var == TypeOf_CoilVSWAHPCoolingEquationFit) ||
-                    (SELECT_CASE_var == TypeOf_CoilVSWAHPHeatingEquationFit) || (SELECT_CASE_var == TypeOf_PackagedTESCoolingCoil)) {
+                if ((SELECT_CASE_var == DataPlant::PlantEquipmentType::CoilWaterCooling) ||
```

Comment:
This absolutely has to be a `switch/case` or some kind of lookup. A conditional with this many clauses is very slow.